### ISSUE TITLE
Problem validate attribute with union type

### ISSUE THREAD

**Issue by avillegasfdz** [brunato/xmlschema#103]
Hi,

I have a XSD file to validate the XML I am generating. I have validated the XML against the XSD using other tools (i.e., Pycharm's validator) and it is correct, but xmlschema.is_valid does not validate it. Sorry I can't post the XSD or XML, I'll try my best to describe the issue.

Description:

In the XSD I have type <xs:simpleType name="myType"> which is a <xs:union memberTypes="TypeA TypeB" />.
TypeA is a simpleType whose base type is base="xs:decimal" with xs:minInclusive, xs:maxInclusive and xs:fractionDigits restrictions;  and TypeB is a simpleType with base="xs:string" and the pattern restriction to be ="NaN".
In summary, myType is used to allow an attribute to be either a decimal number within a range or NaN.

When parsing a XML (which, as I said, is valid according to other tools), and setting an attribute of type myType to "NaN", I receive the following error:   
File "[...] python3.6.3/lib/python3.6/site-packages/xmlschema/validators/facets.py", line 280, in min_inclusive_validator
    if x < self.value:
decimal.InvalidOperation: [<class 'decimal.InvalidOperation'>]

It seems that the "NaN" string is parsed as xs:decimal instead of xs:string, and then is not able to validate that the minInclusive restriction.
(EDIT: I have found the XML specification for xs:decimal and NaN should not be parsed as decimal -
https://www.w3.org/TR/xmlschema-2/#decimal-lexical-representation )

However, I have found that if I modify the XSD file and change the definition of myType from <xs:union memberTypes="TypeA TypeB" /> to <xs:union memberTypes="TypeB TypeA" /> (this is, I write the NaN type first and the decimal type second), the file is validated correctly.

I am not sure if, according to the XML 1.0 specification, the order of the members in the xs:union  should affect the outcome of the validation or not (and, thus, I should change the XSD specification), or if it is a bug. 
(EDIT: According to the XML specification, the members of the union are processed in order: 4.1.2.3 Derivation by union from https://www.w3.org/TR/xmlschema-2/
As a workaround I can set the NaN type as first member in the union but, according to the specification, NaN should not be parsed as decimal)

Thanks.

**Comment by brunato** [comment]
Hi,

maybe fixed in release v1.0.11. As a workaround I've added an extra validator function to xs:decimal:

```python
def finite_number_validator(x):
    try:
        if isinf(x) or isnan(x):
            yield XMLSchemaValidationError(finite_number_validator, x, "value {!r} is not an xs:decimal".format(x))
    except TypeError:
        pass
```

Try the new release and it it works please report me or close this issue.
Thanks

**Comment by avillegasfdz** [comment]
Hi,

It works on my sample XML files that include NaN as decimal number. 
Thanks for the fix. Star earned!
